presto-kafka/src/main/java/com/facebook/presto/kafka/encoder/raw/RawRowEncoder.java

```diff
@@ -0,0 +1,329 @@
+            }
+            else if (columnType == TINYINT) {
+                checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.BYTE);
+            }
+            else if (columnType == BOOLEAN) {
+                checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.BYTE, FieldType.SHORT, FieldType.INT, FieldType.LONG);
+            }
+            else if (columnType == DOUBLE) {
+                checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.DOUBLE, FieldType.FLOAT);
+            }
+            else if (isVarcharType(columnType)) {
+                checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.BYTE);
+            }
+        }
+
+        private static void checkFieldTypeOneOf(FieldType declaredFieldType, String columnName, Type columnType, FieldType... allowedFieldTypes)
+        {
+            checkArgument(Arrays.asList(allowedFieldTypes).contains(declaredFieldType),
+                    format("Wrong dataformat '%s' specified for column '%s'; %s type implies use of %s",
+                            declaredFieldType.name(),
+                            columnName,
+                            columnType,
+                            Joiner.on("/").join(allowedFieldTypes)));
+        }
+
+        public String getName()
+        {
+            return name;
+        }
+
+        public Type getType()
+        {
+            return type;
+        }
+
+        public int getStart()
+        {
+            return start;
+        }
+
+        public int getEnd()
+        {
+            return end;
+        }
+
+        public FieldType getFieldType()
+        {
+            return fieldType;
+        }
+
+        public int getLength()
+        {
+            return end - start;
+        }
+    }
+
+    private static boolean isSupportedType(Type type)
+    {
+        return isVarcharType(type) || SUPPORTED_PRIMITIVE_TYPES.contains(type);
+    }
+
+    @Override
+    protected void appendLong(long value)
+    {
+        buffer.putLong(value);
+    }
+
+    @Override
+    protected void appendInt(int value)
+    {
+        buffer.putInt(value);
+    }
+
+    @Override
+    protected void appendShort(short value)
+    {
+        buffer.putShort(value);
+    }
+
+    @Override
+    protected void appendByte(byte value)
+    {
+        buffer.put(value);
+    }
+
+    @Override
+    protected void appendDouble(double value)
+    {
+        buffer.putDouble(value);
+    }
+
+    @Override
+    protected void appendFloat(float value)
+    {
+        buffer.putFloat(value);
+    }
+
+    @Override
+    protected void appendBoolean(boolean value)
+    {
+        buffer.put((byte) (value ? 1 : 0));
+    }
+
+    @Override
+    protected void appendString(String value)
+    {
+        byte[] valueBytes = value.getBytes(StandardCharsets.UTF_8);
+        checkArgument(valueBytes.length == columnMappings.get(currentColumnIndex).getLength(), format(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;getLength&#39; expression, valueBytes.length. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 298. Consider formatting this data with &#39;%d&#39; instead.